modules/develop/pages/consume-data/consumer-offsets.adoc

```diff
@@ -4,20 +4,125 @@
+== Commit strategies
+
+There are multiple commit strategies available. The most common follow.
+
+=== Auto commit
+
+Auto commit is the default commit strategy, where the client automatically commits offsets at regular intervals. You can enable this by setting `enable.auto.commit` to `true`. The client will then commit offsets every `auto.commit.interval.ms` milliseconds.
+
+The primary advantage of the auto commit approach is its simplicity. Once configured, the consumer requires no additional effort, as commits are managed in the background. However, this "hands-off" approach has a drawback: the consumer is unaware of what was committed or when. As a result, after an application restart, some messages may be reprocessed (since consumption resumes from the last committed offset, which may include already processed messages). The behavior provides an “at least once” delivery guarantee.
+
+=== Manual offset commit
+
+The manual offset commit strategy gives consumers greater control over when commits occur. This approach is typically used when a consumer needs to align commits with an external system, such as database transactions in an RDBMS. The main advantage of manual commits is that they allow you to decide exactly when a record is considered consumed. You can use two API calls for this: `commitSync` and `commitAsync`, which differ in their blocking behavior.
+
+==== Synchronous commit
+
+The advantage of synchronous commits is that consumers can take appropriate action before continuing to consume messages, albeit at the expense of increased latency (while waiting for the commit to return). The commit (`commitSync`) will also retry automatically, until it either succeeds or receives an unrecoverable error. The following pseudocode shows an example of synchronous commit:
+
+[source,java]
+----
+consumer.subscribe(Arrays.asList("foo", "bar"));
+while (true) {
+    ConsumerRecords<String, String> records = consumer.poll(100);
+    for (ConsumerRecord<String, String> record : records) {
+        // process records here ...
+        
+        // ... and at the appropriate point, call commit (not after every message)
+        consumer.commitSync();
+    }
+}
+----
+
+==== Asynchronous commit
+
+The advantage of asynchronous commits is lower latency, because the consumer does not pause to wait for the commit response. However, a disadvantage is that there is no automatic retry of the commit (`commitAsync`) if it fails. There is also increased coding complexity (due to the asynchronous callbacks) to consider. The following pseudocode shows an example of an asynchronous commit in which the consumer will not block. Instead, the commit call registers a callback, which is executed once the commit returns:
```

Comment:
```suggestion
The advantage of asynchronous commits is lower latency, because the consumer does not pause to wait for the commit response. However, there is no automatic retry of the commit (`commitAsync`) if it fails. There is also increased coding complexity (due to the asynchronous callbacks). The following pseudocode shows an example of an asynchronous commit in which the consumer will not block. Instead, the commit call registers a callback, which is executed once the commit returns:
```